src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -3578,6 +3578,22 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
             if (ASRUtils::is_integer(*t) ||
                 ASR::is_a<ASR::Logical_t>(*ASRUtils::type_get_past_pointer(t))) {
                 switch( a_kind ) {
+                    case 1 : {
+                        if (llvm::ConstantInt* CI = llvm::dyn_cast<llvm::ConstantInt>(tmp)) {
+                            if (CI->getBitWidth() <= 8) {
+                                if (CI->isZero())
+                                {
+                                    tmp = builder->CreateGlobalStringPtr("False");
+                                }
+                                else if (CI->isOne())
+                                {
+                                    tmp = builder->CreateGlobalStringPtr("True");
+                                }
+                            }
+                        }
+                        fmt.push_back("%s");
+                        break;
+                    } 
```

Comment:
This will not work for `i: i8`, which is an integer of `kind=1`. What we need here is to separate integer printing form logical printing, and integer printing should print for all kinds (1, 2, 4, 8) as a number, while logical printing should print True/False.